bundles/org.openhab.binding.homewizard/src/main/java/org/openhab/binding/homewizard/internal/handler/HomeWizardP1MeterHandler.java

```diff
@@ -87,40 +93,39 @@ protected void handleDataPayload(DataPayload payload) {
         updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_POWER_L3,
                 new QuantityType<>(payload.getActivePowerL3W(), Units.WATT));
 
-        // If no data from the gas meter is present, the json value will be null, which means gson ignores it,
-        // leaving the value in the payload object at 0.
-        long dtv = payload.getGasTimestamp();
-        if (dtv > 0) {
-            updateState(HomeWizardBindingConstants.CHANNEL_TOTAL_GAS,
+        updateState(HomeWizardBindingConstants.CHANNEL_POWER_FAILURES, new DecimalType(payload.getAnyPowerFailCount()));
+        updateState(HomeWizardBindingConstants.CHANNEL_LONG_POWER_FAILURES,
+                new DecimalType(payload.getLongPowerFailCount()));
+
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_CURRENT,
+                new QuantityType<>(payload.getActiveCurrent(), Units.AMPERE));
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_CURRENT_L1,
+                new QuantityType<>(payload.getActiveCurrentL1(), Units.AMPERE));
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_CURRENT_L2,
+                new QuantityType<>(payload.getActiveCurrentL2(), Units.AMPERE));
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_CURRENT_L3,
+                new QuantityType<>(payload.getActiveCurrentL3(), Units.AMPERE));
+
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_VOLTAGE,
+                new QuantityType<>(payload.getActiveVoltage(), Units.VOLT));
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_VOLTAGE_L1,
+                new QuantityType<>(payload.getActiveVoltageL1(), Units.VOLT));
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_VOLTAGE_L2,
+                new QuantityType<>(payload.getActiveVoltageL2(), Units.VOLT));
+        updateState(HomeWizardBindingConstants.CHANNEL_ACTIVE_VOLTAGE_L3,
+                new QuantityType<>(payload.getActiveVoltageL3(), Units.VOLT));
+
+        ZonedDateTime gasTimestamp;
+        try {
+            gasTimestamp = payload.getGasTimestamp(timeZoneProvider.getTimeZone());
+        } catch (DateTimeException e) {
+            logger.warn("Unable to parse Gas timestamp: {}", e.getMessage());
+            gasTimestamp = null;
+        }
+        if (gasTimestamp != null) {
```

Comment:
Just checking, it shouldn't be cleared (`UNDEF`) otherwise?

Reply:
Can we get this in m2 ?